owncloudApp/src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -225,27 +226,55 @@ public void onActivityCreated(Bundle savedInstanceState) {
 
         if (savedInstanceState != null) {
             mFile = savedInstanceState.getParcelable(KEY_FILE);
+            mFileListOption = savedInstanceState.getParcelable(KEY_FILE_LIST_OPTION);
+        }
+
+        if (mFileListOption == null && getArguments() != null && getArguments().getParcelable(ARG_LIST_FILE_OPTION) != null) {
```

Comment:
```suggestion
        if (mFileListOption == null) {
            if (getArguments() != null && getArguments().getParcelable(ARG_LIST_FILE_OPTION) != null) {
                mFileListOption = getArguments().getParcelable(ARG_LIST_FILE_OPTION);
            }
            mFileListOption = FileListOption.ALL_FILES;
        }
```

Reply:
You can select several lines when you comment, that way the suggestion would affect all the lines you would change and not just the first. I'll take a look 👍

Reply:
It is not exactly the same, mFileListOption would be always ALL_FILES with that solution. With kotlin it would be less verbose.

Reply:
You are right, let's wait for the kotlin solution then